### PR TITLE
support variable to enable ws ping pong for jb transcribe

### PR DESCRIPTION
**Pull request by xquanluu**
https://github.com/jambonz/freeswitch-modules/issues/94

### CODE REVIEW

**Comment by xquanluu** [comment]
Tested on jambonz.me:

1/ ping was sent every 10 seconds
2/ without pong, ws would be closed after 30 seconds

File: mod_jambonz_transcribe/audio_pipe.cpp [hunk 1]
```diff
@@ -230,13 +230,16 @@ int AudioPipe::lws_callback(struct lws *wsi,
 
 
 // static members
+const bool ping_pong_enabled = switch_true(std::getenv("WS_PING_PONG_ENABLED"));
```
Comment (davehorton) [inline mod_jambonz_transcribe/audio_pipe.cpp:233]: lets not use switch_true() and therefore remove the dependency on switch.h

File: mod_jambonz_transcribe/audio_pipe.cpp [hunk 2]
```diff
@@ -230,13 +230,16 @@ int AudioPipe::lws_callback(struct lws *wsi,
...
-    0,         // retry_ms_table_count
-    0,         // conceal_count
-    UINT16_MAX,         // secs_since_valid_ping
-    UINT16_MAX,        // secs_since_valid_hangup
-    0          // jitter_percent
+    nullptr,// retry_ms_table
+    0,// retry_ms_table_count
+    0,// conceal_count
+    // secs_sinceq_valid_ping, if enabled, send ping after connection is idle for 10 seconds
+    ping_pong_enabled ? static_cast<uint16_t>(10) : static_cast<uint16_t>(UINT16_MAX),
```
Comment (davehorton) [inline mod_jambonz_transcribe/audio_pipe.cpp:239]: let change the env var from WS_PING_PONG_ENABLED to WS_PING_INTERVAL, and when the variable is not defined or is zero set to UINT16_MAX

File: mod_jambonz_transcribe/audio_pipe.cpp [hunk 3]
```diff
@@ -230,13 +230,16 @@ int AudioPipe::lws_callback(struct lws *wsi,
...
+    // secs_since_valid_hangup, if enabled, kill the connection after 30 seconds without response pong
+    ping_pong_enabled ? static_cast<uint16_t>(30) : static_cast<uint16_t>(UINT16_MAX),
```
Comment (davehorton) [inline mod_jambonz_transcribe/audio_pipe.cpp:241]: lets have another env var for this WS_NO_PONG_HANGUP_INTERVAL

**Review by davehorton — CHANGES_REQUESTED** [review]

File: mod_jambonz_transcribe/audio_pipe.cpp (hunk 2 above)
Comment (xquanluu) [reply mod_jambonz_transcribe/audio_pipe.cpp]: fixed

File: mod_jambonz_transcribe/audio_pipe.cpp (hunk 3 above)
Comment (xquanluu) [reply mod_jambonz_transcribe/audio_pipe.cpp]: fixed

**Comment by vdharashive** [comment]
@xquanluu and @davehorton this is working as expected

**Review by davehorton — APPROVED** [review]

**Pull request merged**